doc/_extensions/zephyr/git_info.py

```diff
@@ -0,0 +1,103 @@
+"""
+Git Info Extension
+##################
+
+Copyright (c) 2023 The Linux Foundation
+SPDX-License-Identifier: Apache-2.0
+
+Introduction
+============
+
+This extension adds a new Jinja filter, ``git_info``, that returns the date and SHA1 of the last
+commit made to a page if this page is managed by Git.
+"""
+
+import os
+import re
+import subprocess
+from datetime import datetime
+from functools import partial
+from pathlib import Path
+from typing import Optional
+from typing import Tuple
+
+from sphinx.application import Sphinx
+from sphinx.util.i18n import format_date
+
+__version__ = "0.1.0"
+
+
+def git_info_filter(app: Sphinx, pagename) -> Optional[Tuple[str, str]]:
+    """Return a tuple with the date and SHA1 of the last commit made to a page.
+
+    Arguments:
+        app {Sphinx} -- Sphinx application object
+        pagename {str} -- Page name
+
+    Returns:
+        Optional[Tuple[str, str]] -- Tuple with the date and SHA1 of the last commit made to the
+        page, or None if the page is not in the repo.
+    """
+
+    if not os.path.isfile(app.env.project.doc2path(pagename)):
+        return None
+
+    for exclude in app.config.vcs_link_exclude:
+        if re.match(exclude, pagename):
+            return None
+
+    found_prefix = ""
+    for pattern, prefix in app.config.vcs_link_prefixes.items():
+        if re.match(pattern, pagename):
+            found_prefix = prefix
+            break
+
+    orig_path = os.path.join(
+        app.config.ZEPHYR_BASE,
+        found_prefix,
+        app.env.project.doc2path(pagename, basedir=False),
+    )
```

Comment:
Mmmm I think I found why pylint is complaining 🙃
I just remembered that I copy-pasted this from vcs_link extension... Will update to make sure there's  no such duplication...